manifests/params.pp

```diff
@@ -18,7 +18,11 @@
 #
 # Sample Usage:
 #
-class apache::params inherits ::apache::version {
+class apache::params (
+ Boolean $scl            = false,
```

Comment:
It is historically the policy of module design to not allow the params.pp to take parameters due to the ease of fragmenting parameters between the base class and the params class from ye olde days.

Since parse order always has the params class coming before the base class, when there is an exception to modify the behavior of params then the canonical pattern is to have a "globals" class that affects params feeding into the base class. Three examples:
- https://github.com/puppetlabs/puppetlabs-postgresql/blob/master/manifests/globals.pp (due to many of the parameters being used by both `postgresql::server` and `postgresql::client` with no base class common between them. Many of these should be in the base class and the globals class should only be used for the repo management)
- https://github.com/voxpupuli/puppet-mongodb/blob/master/manifests/globals.pp (you can see that this is already starting to have parameter creep even WITH a base class)

Since these three parameters do seem to be repo related, it would be best to create an `apache::globals` class for them.

Reply:
I will move them to globals. This is a cleaner design.

But I think we should also look at: https://docs.puppet.com/puppet/4.3/lookup_quick_module.html
But this is a lot larger task...